File: app/src/main/java/com/cooltechworks/sample/utils/DemoConfiguration.java
package com.cooltechworks.sample.utils;


import androidx.annotation.LayoutRes;
import androidx.annotation.StringRes;
import androidx.annotation.StyleRes;
import androidx.recyclerview.widget.RecyclerView;

public class DemoConfiguration {
    @StyleRes
    private int styleResource;

    @LayoutRes
    private int layoutResource;

    @StringRes
    private int titleResource;

    private RecyclerView.LayoutManager layoutManager;

    private RecyclerView.ItemDecoration itemDecoration;

    public RecyclerView.ItemDecoration getItemDecoration() {
        return itemDecoration;
    }

    public void setItemDecoration(RecyclerView.ItemDecoration itemDecoration) {
        this.itemDecoration = itemDecoration;
    }

    public int getStyleResource() {
        return styleResource;
    }

    public void setStyleResource(int styleResource) {
        this.styleResource = styleResource;
    }

    public int getLayoutResource() {
        return layoutResource;
    }

    public void setLayoutResource(int layoutResource) {
        this.layoutResource = layoutResource;
    }

    public int getTitleResource() {
        return titleResource;
    }

    public void setTitleResource(int titleResource) {
        this.titleResource = titleResource;
    }

    public RecyclerView.LayoutManager getLayoutManager() {
        return layoutManager;
    }

    public void setLayoutManager(RecyclerView.LayoutManager layoutManager) {
        this.layoutManager = layoutManager;
    }
}

File: app/src/main/java/com/cooltechworks/sample/utils/DemoConfiguration.kt
package com.cooltechworks.sample.utils

import androidx.annotation.LayoutRes
import androidx.annotation.StringRes
import androidx.annotation.StyleRes
import androidx.recyclerview.widget.RecyclerView

class DemoConfiguration {
    @StyleRes
    var styleResource: Int = 0

    @LayoutRes
    var layoutResource: Int = 0

    @StringRes
    var titleResource: Int = 0

    var layoutManager: RecyclerView.LayoutManager? = null

    var itemDecoration: RecyclerView.ItemDecoration? = null
}
